components/ListView/ComponentsDataListItem.js

```diff
@@ -0,0 +1,217 @@
+import React from "react";
+import { defineMessages, injectIntl, intlShape, FormattedMessage } from "react-intl";
+import PropTypes from "prop-types";
+import {
+  DataListItem,
+  DataListItemRow,
+  DataListToggle,
+  DataListCell,
+  DataListItemCells,
+  DataListContent
+} from "@patternfly/react-core";
+import ComponentTypeIcons from "./ComponentTypeIcons";
+import ComponentSummaryList from "./ComponentSummaryList";
+
+const messages = defineMessages({
+  details: {
+    defaultMessage: "details"
+  },
+  actions: {
+    defaultMessage: "actions"
+  }
+});
+
+class ComponentsDataListItem extends React.Component {
+  constructor() {
+    super();
+    this.state = { expanded: false };
+    this.handleExpandComponent = this.handleExpandComponent.bind(this);
+  }
+
+  componentWillReceiveProps(newProps) {
```

Comment:
ComponentWIllReceiveProps is going to be deprecated. Can you change this to a different lifecycle method?